src/cluster_legacy.c

```diff
@@ -1348,9 +1348,9 @@ void clusterAutoFailoverOnShutdown(void) {
 }
 
 /* Called when a cluster node receives SHUTDOWN. */
-void clusterHandleServerShutdown(void) {
+void clusterHandleServerShutdown(bool auto_failover) {
     /* Check if we are able to do the auto failover on shutdown. */
-    clusterAutoFailoverOnShutdown();
+    if (auto_failover) clusterAutoFailoverOnShutdown();
```

Comment:
nitpicks on names

1. `auto` isn't necessary imo - it is implied that the code is doing the work
2. the original function `clusterHandleServerShutdown` doesn't know where/why it is invoked. it just merely provides the failover functionality. 

```suggestion
    if (failover) clusterInitiateGracefulFailover();
```

Reply:
True. This was added in #1091. I removed "auto" from the configs (public API) but I didn't change the internals... I'll see if I can get to it.

Reply:
I guess we can ignore this nitpick and merge, to get it off the table.